app/models/npq_contract.rb

```diff
@@ -6,4 +6,8 @@ class NPQContract < ApplicationRecord
   validates :number_of_payment_periods, :output_payment_percentage, :service_fee_installments, :service_fee_percentage, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
   validates :per_participant, numericality: { greater_than: 0 }
   validates :recruitment_target, numericality: { only_integer: true, greater_than: 0 }
+
+  def course
+    NPQCourse.find_by(identifier: course_identifier)
+  end
```

Comment:
I'd be inclined to make this an actual relationship:
```ruby
belong_to :npc_course, foreign_key: :identifier, class_name: "NPQCourse"
```
That would allow to eager load, join and etc...